lib/enumerable/inject.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_inject(initial = nil, operation = nil)
+    return multiply_els(self, initial) if operation == :*
+
+    verified = verify_input(initial)
+
+    input = self.class == Range ? to_a : self
+    result = initial.nil? || initial.is_a?(Symbol) ? 0 : initial
+    if block_given?
+      start = verified[1]
+      input[start..- 1].my_each do |i|
+        result = yield(result, i)
+      end
+    else
+      input[1..-1].my_each do |i|
+        result = result.send(initial, i)
+      end
+    end
```

Comment:
I think it would be better not to leave out the first element in cases where only a `Symbol` is passed.

```ruby
[1,2,3].inject(:+)   #=> 6
[1,2,3].my_inject(:+) #=> 5
```

Reply:
Done
![image](https://user-images.githubusercontent.com/36519478/68699682-96e06680-0551-11ea-9f70-257b357b95a7.png)